src/ApiClient.php

```diff
@@ -228,17 +305,34 @@ private function fetchQuotes(array $symbols)
     {
         $qs = $this->getRandomQueryServer();
 
-        // Initialize session cookies
-        $cookieJar = $this->getCookies();
+        $crumb = $this->getCrumb();
 
-        // Get crumb value
-        $crumb = $this->getCrumb($qs, $cookieJar);
+        $url = 'https://query'.$qs.'.finance.yahoo.com/v7/finance/quote?symbols='.urlencode(implode(',', $symbols)).'&crumb='.$crumb;
 
-        // Fetch quotes
-        $url = 'https://query'.$qs.'.finance.yahoo.com/v7/finance/quote?crumb='.$crumb.'&symbols='.urlencode(implode(',', $symbols));
-        $responseBody = (string) $this->client->request('GET', $url, ['cookies' => $cookieJar, 'headers' => $this->getHeaders()])->getBody();
+        try {
+            $responseBody = (string) $this->client->request('GET', $url, [
+                'cookies' => $this->getCookies(),
+                'headers' => $this->getHeaders(),
+            ])->getBody();
+
+            return $this->resultDecoder->transformQuotes($responseBody);
+        } catch (\GuzzleHttp\Exception\ClientException $e) {
```

Comment:
As mentioned before, the `ClientException` isn't thrown for 401s, when you have `'http_errors' => false` set.